v-next/hardhat-errors/src/errors.ts

```diff
@@ -76,7 +80,7 @@ export class HardhatError<
           );
 
     super(
-      prefix + formattedMessage,
+      `${errorCode}: ${formattedMessage}`,
```

Comment:
I'm not sure if it's still worth passing he `prefix` here. Using only the `formattedMessage` as the `message` may achieve a cleaner look.

Reply:
I think keeping it is pretty important because the formatting is only done in hardhat's CLI, but these errors can be thrown in other places (e.g., a script using hardhat).

re test failures: I fixed that in #5460